llvm/lib/Transforms/Scalar/Sink.cpp

```diff
@@ -144,9 +145,10 @@ static bool SinkInstruction(Instruction *Inst,
       SuccToSinkTo = DT.findNearestCommonDominator(SuccToSinkTo, UseBlock);
     else
       SuccToSinkTo = UseBlock;
-    // The current basic block needs to dominate the candidate.
-    if (!DT.dominates(BB, SuccToSinkTo))
-      return false;
+    // The current basic block dominates all uses, thus it must dominate
+    // SuccToSinkTo, the nearest common dominator of all uses.
+    assert(DT.dominates(BB, SuccToSinkTo) &&
+           "SuccToSinkTo must be dominated by current basic block!");
```

Comment:
I think it would be better to replace the two asserts with a single one at the end of this function.

Reply:
I've replaced the two asserts with one assert, but I've put the assert at the beginning of the function since it is already true at that point.

Reply:
The reason why I suggest to put it at the end is that SuccToSinkTo is changed in the function. So we are asserting that the property still holds after these adjustments.

Reply:
Moved assert to when SuccToSinkTo is used to sink instruction.